exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,226 @@
+  };
 
   for (var i = 0; i < 50; i++) {
-    popQuestions.push('Pop Question ' + i)
-    scienceQuestions.push('Science Question ' + i)
-    sportsQuestions.push('Sports Question ' + i)
-    rockQuestions.push(this.createRockQuestion(i))
+    popQuestions.push(createQuestion("Pop", i));
+    scienceQuestions.push(createQuestion("Science", i));
+    sportsQuestions.push(createQuestion("Sports", i));
+    rockQuestions.push(createQuestion("Rock", i));
   }
 
-  this.isPlayable = function (howManyPlayers) {
-    return howManyPlayers >= 2
-  }
+  this.checkIsPlayable = function (howManyPlayers) {
+    const minimumPlayers = 2;
+    return howManyPlayers >= minimumPlayers;
+  };
+
+  this.addPlayer = function (playerName) {
+    players.push(playerName);
+    this.setLocation(this.getLastPlayer(), 0);
+    this.setGoldCoins(this.getLastPlayer(), 0);
+    this.setPenaltyBox(this.getLastPlayer(), false);
+
+    console.log(playerName + " was added");
+    console.log("They are player number " + this.getNumberOfPlayers());
+
+    return true;
+  };
+
+  const askQuestion = () => {
+    switch (currentCategory()) {
+      case "Pop":
+        console.log(popQuestions.shift());
+        break;
+      case "Science":
+        console.log(scienceQuestions.shift());
+        break;
+      case "Sports":
+        console.log(sportsQuestions.shift());
+        break;
+      case "Rock":
+        console.log(rockQuestions.shift());
+        break;
+    }
+  };
 
-  this.add = function (playerName) {
-    players.push(playerName)
-    places[this.howManyPlayers() - 1] = 0
-    purses[this.howManyPlayers() - 1] = 0
-    inPenaltyBox[this.howManyPlayers() - 1] = false
+  const movePlayer = (roll) => {
+    this.setLocation(currentPlayer, getCurrentLocation() + roll);
+    if (getCurrentLocation() > 11) {
+      this.setLocation(currentPlayer, getCurrentLocation() - boardSize);
+    }
 
-    console.log(playerName + ' was added')
-    console.log('They are player number ' + players.length)
+    console.log(
+      players[currentPlayer] + "'s new location is " + places[currentPlayer]
+    );
+    console.log("The category is " + currentCategory());
+    askQuestion();
+  };
 
-    return true
-  }
+  this.getInPenaltyBox = function () {
+    return inPenaltyBox[currentPlayer];
+  };
 
-  this.howManyPlayers = function () {
-    return players.length
-  }
+  this.roll = function (roll) {
+    console.log(players[currentPlayer] + " is the current player");
+    console.log("They have rolled a " + roll);
+    const rollIsOddNumber = roll % 2 !== 0;
 
-  var askQuestion = function () {
-    if (currentCategory() == 'Pop') { console.log(popQuestions.shift()) }
-    if (currentCategory() == 'Science') { console.log(scienceQuestions.shift()) }
-    if (currentCategory() == 'Sports') { console.log(sportsQuestions.shift()) }
-    if (currentCategory() == 'Rock') { console.log(rockQuestions.shift()) }
-  }
+    if (this.getInPenaltyBox()) {
+      if (roll % 2 !== 0) {
```

Comment:
Could this nested if be avoided by checking both conditions in the first if? Not sure if this is right but may be worth looking at :)